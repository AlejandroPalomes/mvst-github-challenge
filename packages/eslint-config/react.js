const { resolve } = require("node:path");

const project = resolve(process.cwd(), "tsconfig.json");

/*
 * This is a custom ESLint configuration for use a library
 * that utilizes React.
 *
 * This config extends the Vercel Engineering Style Guide.
 * For more information, see https://github.com/vercel/style-guide
 *
 */

module.exports = {
  extends: [
    "@vercel/style-guide/eslint/browser",
    "@vercel/style-guide/eslint/typescript",
    "@vercel/style-guide/eslint/react",
  ].map(require.resolve),
  parserOptions: {
    project,
  },
  plugins: ["only-warn"],
  globals: {
    JSX: true,
  },
  settings: {
    "import/resolver": {
      typescript: {
        project,
      },
    },
  },
  ignorePatterns: ["node_modules/", "dist/", ".eslintrc.js", "**/*.css"],
  // add rules configurations here
  rules: {
    "import/no-default-export": "off",
    "@typescript-eslint/explicit-function-return-type": "off",
    "jsx-a11y/click-events-have-key-events": "off",
    "no-console": "off",
    "@typescript-eslint/no-unsafe-member-access": "off",
    "@typescript-eslint/no-explicit-any": "off",
    "react/jsx-no-leaked-render": "off",
    "jsx-a11y/no-static-element-interactions": "off",
    "react/function-component-definition": [2, { "namedComponents": "arrow-function" }]
  },
};
