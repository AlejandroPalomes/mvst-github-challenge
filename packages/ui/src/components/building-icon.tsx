import { type FC } from 'react';
import { type IconProps, getStrokeWidth } from '../utils/icon-utils';

export const BuildingIcon: FC<IconProps> = ({ color = 'white', size = 24}) => {
  const strokeWidth = getStrokeWidth(size);

  return (
    <svg
      fill="none"
      height={size}
      viewBox="0 0 24 24"
      width={size}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M14.4744 7.54607V4.57677C14.4744 4.03014 14.0312 3.58701 13.4846 3.58701H4.57668C4.03005 3.58701 3.58691 4.03014 3.58691 4.57677V19.4233C3.58691 19.9699 4.03005 20.413 4.57668 20.413H14.4744V18.4335M12.4948 10.5154V11.5051M12.4948 14.4744V15.4642M16.4539 10.5154V11.5051M16.4539 14.4744V15.4642M8.53575 20.413H19.4232C19.9698 20.413 20.413 19.9699 20.413 19.4233V8.53584C20.413 7.98921 19.9698 7.54607 19.4232 7.54607H9.52552C8.97888 7.54607 8.53575 7.98921 8.53575 8.53584V20.413Z"
        stroke={color}
        strokeLinejoin="round"
        strokeWidth={strokeWidth}
      />
    </svg>
  );
};
