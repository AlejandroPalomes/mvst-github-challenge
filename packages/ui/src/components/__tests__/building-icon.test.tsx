import { render } from "@testing-library/react";
import { BuildingIcon } from "../building-icon";

describe('BuildingIcon test', () => {
  it('should render without crashing', () => {
    const { baseElement } = render(<BuildingIcon/>);
    expect(baseElement).toMatchSnapshot();
  });
})