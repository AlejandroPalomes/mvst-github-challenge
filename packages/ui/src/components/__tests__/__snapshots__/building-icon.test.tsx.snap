// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BuildingIcon test > should render without crashing 1`] = `
<body>
  <div>
    <svg
      fill="none"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M14.4744 7.54607V4.57677C14.4744 4.03014 14.0312 3.58701 13.4846 3.58701H4.57668C4.03005 3.58701 3.58691 4.03014 3.58691 4.57677V19.4233C3.58691 19.9699 4.03005 20.413 4.57668 20.413H14.4744V18.4335M12.4948 10.5154V11.5051M12.4948 14.4744V15.4642M16.4539 10.5154V11.5051M16.4539 14.4744V15.4642M8.53575 20.413H19.4232C19.9698 20.413 20.413 19.9699 20.413 19.4233V8.53584C20.413 7.98921 19.9698 7.54607 19.4232 7.54607H9.52552C8.97888 7.54607 8.53575 7.98921 8.53575 8.53584V20.413Z"
        stroke="white"
        stroke-linejoin="round"
        stroke-width="1.5"
      />
    </svg>
  </div>
</body>
`;
